int/integration.go

```diff
@@ -186,3 +187,41 @@ func (i *Integration) WaitForPrometheusRuleToClear(name types.NamespacedName, se
 	}
 	return nil
 }
+
+func (i *Integration) WaitForPrometheusRuleCorrectSLO(name types.NamespacedName, targetSlo string, seconds int) error {
+	prometheusRule := monitoringv1.PrometheusRule{}
+	err := i.Client.Get(context.TODO(), name, &prometheusRule)
+	if errors.IsNotFound(err) {
+		return fmt.Errorf("PrometheusRule wasn't found")
+	}
+	if err != nil {
+		return err
+	}
+
+	routeMonitor := v1alpha1.RouteMonitor{}
+	err = i.Client.Get(context.TODO(), name, &routeMonitor)
+	if errors.IsNotFound(err) {
+		return fmt.Errorf("RouteMonitor wasn't found")
+	}
+	if err != nil {
+		return err
+	}
+
+	template := templates.TemplateForPrometheusRuleResource(routeMonitor.Status.RouteURL, targetSlo, name)
```

Comment:
Why should we create the template of we have the routemonitors before any action was done?

Reply:
Because inside the loop we're going to compare the actual PrometheusRule to the template, to check and see if it is being updated. I didn't want to generate the template from within the loop to avoid calling the "TemplateForPrometheusRuleResource" multiple times for no reason.